apps/web/src/pages/DirectiveForms/CompoundQuestion.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+
+export type CompoundQuestionProps = {
+  title?: string;
+  introduction?: React.ReactElement | React.ReactNode | string;
+  inputElement: React.ReactElement;
+};
+
+// This component is for adding extra information before a form input and ensuring that the aria labels are properly attached.
+const CompoundQuestion = ({
+  title,
+  introduction,
+  inputElement,
+}: CompoundQuestionProps) => {
+  const descriptionId = React.useId();
+  const detailsId = React.useId();
+  const clonedInput = React.cloneElement(inputElement, {
+    "aria-describedby": title ? descriptionId : undefined,
+    "aria-details": introduction ? detailsId : undefined,
+  });
+  return (
+    <div>
+      <div data-h2-margin-top="base(x.5)">
+        {title ? (
+          <div data-h2-font-weight="base(700)" id={descriptionId}>
+            {title}
+          </div>
+        ) : null}
+        {introduction ? (
+          <div data-h2-margin-top="base(x.5)" id={detailsId}>
+            {introduction}
+          </div>
+        ) : null}
```

Comment:
An example of this, for consideration:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/a1bbdc31-1d14-4a52-aafd-626e6e1f7ade)

Reply:
Response from Rosita: yes, whole list should be part of description

Reply:
OK, I think this is working.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/1de8ac3e-37a8-43d1-83a0-777c3a1a4b86)
Ref: [Add details to description](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7771/commits/17bd5b3d39740478944e30be3bd99f94e07da6f6)

Reply:
So, this works very well with single inputs. However, with fieldsets (like checkboxes and radio groups) it does not read out the description while tabbing through inputs.

I'm not sure what approach we want to take here 😖 

This is specifically a problem for technological change factors where the list needs to be read out to understand the context. Would it be possible to instead use a checklist and ask users to select all that apply instead of "yes/no" radio group?

![Screenshot 2023-10-13 124017](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/638d949f-8f47-4c27-bffd-b253c313d141)

Reply:
Discussed in scrum - spun off to its own issue. #8230